pkgs/development/libraries/audio/roc-toolkit/default.nix

```diff
@@ -0,0 +1,54 @@
+{ stdenv, lib, fetchFromGitHub, sconsPackages, ragel, gengetopt, pkg-config,
+  libuv,
+  openfec ? null,
+  libunwind ? null,
+  libpulseaudio ? null
+}:
+
+stdenv.mkDerivation rec {
+  pname = "roc-toolkit";
+  version = "0.1.5";
+
+  src = fetchFromGitHub {
+    owner = "roc-streaming";
+    repo = "roc-toolkit";
+    rev = "v${version}";
+    sha256 = "sha256:1pld340zfch4p3qaf5anrspq7vmxrgf9ddsdsq92pk49axaaz19w";
+  };
+
+  nativeBuildInputs = [
+    sconsPackages.scons_3_0_1
+    ragel
+    gengetopt
+    pkg-config
+  ];
+
+  buildInputs = [
+    libuv
+    libunwind
+    openfec
+    libpulseaudio
+  ];
+
+  sconsFlags =
+    [ "--disable-sox"
+      "--disable-tests" ] ++
+    lib.optional (libunwind == null) "--disable-libunwind" ++
+    lib.optional (libpulseaudio == null) "--disable-pulseaudio" ++
+    (if openfec == null
+       then ["--disable-openfec"]
+       else [ "--with-libraries=${openfec}/lib"
+              "--with-openfec-includes=${openfec.dev}/include" ]);
+
+  preConfigure = ''
```

Comment:
Is there a reason why this isn't simply part of `sconsFlags`? Also consider [doing what the godot derivation is doing for scons options](https://github.com/NixOS/nixpkgs/blob/db2ad441370111db38a7d5e52cf74857a0502860/pkgs/development/tools/godot/default.nix#L71).

Reply:
Handling the nix side of things might be cleaner that way, but it is your call. If that method can't handle all options needed here then the benefits are indeed dubious.